org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DeltaProcessor.java

```diff
@@ -2022,7 +2023,8 @@ public void resetProjectCaches() {
 	 */
 	public void registerJavaModelDelta(IJavaElementDelta delta) {
 		if (JavaModelManager.isReadOnly()) {
-			throw new IllegalStateException("Its not allow to modify JavaModel during ReadOnly action. delta=" + delta); //$NON-NLS-1$
+			ILog.get().warn("JavaModel change during readOnly Operation", new IllegalStateException( //$NON-NLS-1$
+					"Its not allow to modify JavaModel during ReadOnly action. delta=" + delta)); //$NON-NLS-1$
```

Comment:
Spelling / wording: 
`Its not allow to modify JavaModel during ReadOnly action`
 -> 
`JavaModel modified during "read only" operation. Consider to report this warning to https://github.com/eclipse-jdt/eclipse.jdt.core/issues`.

Reply:
done